airflow/cli/commands/task_command.py

```diff
@@ -42,9 +45,29 @@
     get_dags,
     suppress_logs_and_warning,
 )
+from airflow.utils.dates import timezone
 from airflow.utils.log.logging_mixin import StreamLogWriter
 from airflow.utils.net import get_hostname
-from airflow.utils.session import create_session
+from airflow.utils.session import create_session, provide_session
+
+
+def _get_ti(dag, task, exec_date_or_run_id):
+    """Get the task instance through DagRun.run_id, if that fails, get the TI the old way"""
+    dag_run = dag.get_dagrun(run_id=exec_date_or_run_id)
+    if not dag_run:
+        try:
+            execution_date = timezone.parse(exec_date_or_run_id)
+            # print("Warning: execution_date will be removed in
+            # tasks command in the future. Please use DagRun.run_id")
+            ti = TaskInstance(task, execution_date)
```

Comment:
I was wondering if it makes sense to add `ti.refresh_from_db()` in to this path, and I think probably yes, even though only one of the tasks _needs_ it, this makes it predictable I think.

We can then remove the call on L250.

Thoughts?

Reply:
Yes. Having it refreshed here is best!